lib/mcb/provider_editor.rb

```diff
@@ -69,7 +171,8 @@ def update(attrs)
     end
 
     def check_authorisation
-      raise Pundit::NotAuthorizedError unless ProviderPolicy.new(@requester, @provider).update?
+      action = @provider.persisted? ? :update? : :create?
+      raise Pundit::NotAuthorizedError unless ProviderPolicy.new(@requester, @provider).send(action)
```

Comment:
Minor improvement, but this use's Pundit's built-in mechanism to find `ProviderPolicy`.

```suggestion
      raise Pundit::NotAuthorizedError unless Pundit.policy(@requester, @provider).send(action)
```

Reply:
fixed and force-pushed.